### PR TITLE
dts: bindings: serial: stm32: block clock-frequency

### LINKED ISSUE #69618: stm32: bindings: serial: clock-frequency should be removed as not supported
**To Reproduce**
`clock-frequency` property appears as a supported property in stm32 serial bindings: 
https://docs.zephyrproject.org/latest/build/dts/api/bindings/serial/st%2Cstm32-usart.html#dtbinding-st-stm32-usart

This property is inherited from `uart-controller.yaml`, but it isn't actually supported as device clock freq is queried directly from clock_control API.

**Expected behavior**
Remove this property from stm32 uart dt bindings, by putting it in `property-blocklist`.

**Impact**
Misleading documentation

### PR DESCRIPTION
**Pull request by GeorgeCGV**
Resolves #69618.

### CODE REVIEW

**Review by dcpleung — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]
Note that blocking this property   `clock-frequency:`  in th st,stm32-uart-base.yaml, will affect 
st,stm32-uart.yaml
st,stm32-usart.yaml
st,stm32-lpuart.yaml
(issue #69618 mainly talks about st,stm32-usart.yaml)

--> please confirm that intention

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by GeorgeCGV** [comment]
> Note that blocking this property `clock-frequency:` in th st,stm32-uart-base.yaml, will affect st,stm32-uart.yaml st,stm32-usart.yaml st,stm32-lpuart.yaml (issue #69618 mainly talks about st,stm32-usart.yaml)
> 
> --> please confirm that intention

The property is not used by `drivers/serial/uart_stm32.c`.

**Review by henrikbrixandersen — DISMISSED** [review]

File: dts/bindings/serial/st,stm32-uart-base.yaml [hunk 1]
```diff
@@ -4,7 +4,12 @@
 # Common fields for STM32 UART peripherals.
 description: STM32 UART-BASE
 
-include: [uart-controller.yaml, pinctrl-device.yaml, reset-device.yaml]
+include:
+  - name: uart-controller.yaml
+    property-blocklist:
+      - clock-frequency
+  - pinctrl-device.yaml
+  - reset-device.yaml
```
Comment (erwango) [inline dts/bindings/serial/st,stm32-uart-base.yaml:12]: Naive question, shouldn't is be
```suggestion
  - name: pinctrl-device.yaml
  - name: reset-device.yaml
```
?

Comment (GeorgeCGV) [reply dts/bindings/serial/st,stm32-uart-base.yaml:12]: The yaml doesn't require that, but it can be done for consistency. Shall I?

Comment (erwango) [reply dts/bindings/serial/st,stm32-uart-base.yaml:12]: For consistency it would be nice indeed (and avoids questions on the next one reading these lines)

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**